Create blog for network isolated cluster GA (#4972)
Azure/AKS: blog/_posts/2025-04-24-network-isolated-cluster.md

---
title: "Network Isolated Cluster"
description: "Use network isolated AKS cluster to restrict outbound traffic at glance"
date: 2025-04-24
author: Charles Wu
categories: security
---

## Network isolated Azure Kubernetes Service (AKS) cluster

> You may have heard about [AKS network isolated cluster](https://learn.microsoft.com/azure/aks/concepts-network-isolated) in the Build last year. This feature allows you to bootstrap an AKS cluster without any outbound egress traffic at glance. This feature is now general available. With this announcement, we want to share with you more details of this exciting offering. 

Reviewer: ```suggestion
You may have heard about [AKS network isolated cluster](https://learn.microsoft.com/azure/aks/concepts-network-isolated) during Microsoft Build last year. This feature allows you to bootstrap an AKS cluster without any outbound egress traffic at a glance. This feature is now general available. With this announcement, we want to share with you more details of this exciting offering. 
```

Reviewer 2: Agree with saying specifically "at Microsoft Build last year". Also might want to change to "Generally available"

Author: I guess we do need the accuracy for propaganda, in coroporated.

> Previously, if you want to restrict outbound traffic for an AKS cluster, firewall is the common option to go with. However, this solution are not perfect for 1. It could be cumbersome and complicated since you have to figure out the endpoints the AKS cluster and your own application require for outbound traffic and to create corresponding egress firewall rules. 2. It is not truly build a cluster without any outbound traffic due to those endpoints a normal AKS cluster requires, these outbound traffic could still cause data exfiltration and leakage. Is there an easier and even safer solution? Gladly, you have network isolated cluster now.

Reviewer 2: "Previously, if a user wants to restrict outbound traffic for an AKS cluster, firewall is the common option to go with."- do we want to say "Azure Firewall" explicitly? 
Maybe reword "Is there an easier and safer solution? With the introduction of network isolated cluster, you have one now." to "Network isolated cluster offers an easier and safer solution."

> Previously, if you want to restrict outbound traffic for an AKS cluster, firewall is the common option to go with. However, this solution are not perfect for 1. It could be cumbersome and complicated since you have to figure out the endpoints the AKS cluster and your own application require for outbound traffic and to create corresponding egress firewall rules. 2. It is not truly build a cluster without any outbound traffic due to those endpoints a normal AKS cluster requires, these outbound traffic could still cause data exfiltration and leakage. Is there an easier and even safer solution? Gladly, you have network isolated cluster now.

Author: Good point for both version, incorporated with both! Tks!!

Network isolated clusters have these benefits:

> - **Zero outbound egress:** A network isolated cluster doesn't require any egress traffic beyond the VNet throughout the cluster bootstrapping process. This is achieved through private endpoint connection to the associated private Azure Container Registry (ACR) instance. Where the later one pulls images and binaries required for AKS cluster creation and operation from the the Microsoft Artifact Registry (MAR). This proxy architecture helps eliminate all outbound requests for AKS cluster bootstrapping and maintenance.

Reviewer: ```suggestion
- **Zero outbound egress:** A network isolated cluster doesn't require any egress traffic beyond the VNet throughout the cluster bootstrapping process. This is achieved through private endpoint connection to the associated private Azure Container Registry (ACR) instance; images and binaries required for AKS cluster creation and operation are pulled from Microsoft Artifact Registry (MAR). This proxy architecture helps eliminate all outbound requests during AKS cluster bootstrapping and maintenance.
```

> - **Simplicity:** Insteading creating a tedious list of firewall rules. You just need to create a network isolated cluster. That's all. Then, you no longer need to worry about the outbound traffic for the cluster creation, operation and upgrading.

Reviewer: ```suggestion
- **Simplicity:** Instead of creating a tedious list of firewall rules, you can just create a network isolated cluster. That's all. With network isolated clusters, you no longer need to worry about the outbound traffic during cluster creation, operation, and upgrading.
```

Reviewer 2: Maybe remove extraneous statements like "That's all"

Author: Just want to call out the simplicity of using network isolated cluster to eliminate outbound traffic (which is quite amount of actions if using Az Firewall). Though I do want to add clarification here it's with managed ACR path.

> - **Secure by default:**  A network isolated cluster reduces the risk of data exfiltration or unintentional exposure of cluster's public endpoints, this is critical if you have strict security and compliance requirements to regulate egress (outbound) network traffic.

Reviewer: ```suggestion
- **Secure by default:**  A network isolated cluster reduces the risk of data exfiltration or unintentional exposure of the cluster's public endpoints. This is critical if you have strict security and compliance requirements to regulate egress (outbound) network traffic.
```

Author: incorporated

> With above being said, firewall still has its own unique value. Especially when you need outbound traffic anyway from your applications or cluster elsewhere and when you want to control, inspect and secure cluster traffic, not only egress but also ingress, at fine grid.

Reviewer: ```suggestion
Firewalls still have their own unique value proposition. This is especially evident when you need outbound traffic from your applications or cluster elsewhere, or when you want to control, inspect, and secure cluster traffic, both egress and ingress, in a finer manner.
```

Reviewer: Maybe adding in this link somewhere for users who get curious about firewalls?
https://learn.microsoft.com/en-us/azure/firewall/protect-azure-kubernetes-service

Author: Incorporated both, I will use this doc for AKS firewall: https://learn.microsoft.com/en-us/azure/aks/limit-egress-traffic?tabs=aks-with-system-assigned-identities

Network isolated cluster is now general available across all Azure public cloud. You can get started today by following the [how-to guide](https://learn.microsoft.com/en-us/azure/aks/network-isolated?pivots=aks-managed-acr) to create a network isolated cluster and secure your cluster environment.

> We are far from the ending, we have a ton of planned experiences that will be shipping over the following months and we'd love to hear your [feedback and suggestions](https://github.com/Azure/AKS/issues/3319) on how to make the network isolated clusters even better.

Reviewer 2: We are far from finished. We have tons of planned experiences that will be shipping over the following months and would love to hear your [feedback and suggestions](https://github.com/Azure/AKS/issues/3319) on how to make  network isolated clusters even better.

Author: Incorporated
